src/pages/AddEditCustomer.tsx

```diff
@@ -0,0 +1,745 @@
+      is_right_of_occupancy_housing_changer: yup.boolean().nullable(),
+      last_contact_date: yup
+        .string()
+        .nullable()
+        .transform((val, originalVal) => (originalVal === '' ? null : val)),
+      right_of_residence: yup
+        .number()
+        .nullable()
+        .transform((val, originalVal) => (originalVal === '' ? null : val)),
+      primary_profile: yup
+        .object({
+          city: yup.string().required(t(`${T_PATH}.required`)),
+          contact_language: yup
+            .string()
+            .oneOf(langCodes)
+            .required(t(`${T_PATH}.required`)),
+          date_of_birth: yup.string().required(t(`${T_PATH}.required`)),
+          email: yup
+            .string()
+            .email()
+            .required(t(`${T_PATH}.required`)),
+          first_name: yup.string().required(t(`${T_PATH}.required`)),
+          last_name: yup.string().required(t(`${T_PATH}.required`)),
+          national_identification_number: yup.string().nullable(),
+          phone_number: yup.string().required(t(`${T_PATH}.required`)),
+          postal_code: yup.string().required(t(`${T_PATH}.required`)),
+          street_address: yup.string().required(t(`${T_PATH}.required`)),
+        })
+        .required(),
+      secondary_profile: yup
+        .object()
+        .when('has_secondary_profile', {
+          is: true,
+          then: yup.object({
+            city: yup.string().required(t(`${T_PATH}.required`)),
+            contact_language: yup
+              .string()
+              .oneOf(langCodes)
+              .required(t(`${T_PATH}.required`)),
+            date_of_birth: yup.string().required(t(`${T_PATH}.required`)),
+            email: yup
+              .string()
+              .email()
+              .required(t(`${T_PATH}.required`)),
+            first_name: yup.string().required(t(`${T_PATH}.required`)),
+            last_name: yup.string().required(t(`${T_PATH}.required`)),
+            national_identification_number: yup.string().nullable(),
+            phone_number: yup.string().required(t(`${T_PATH}.required`)),
+            postal_code: yup.string().required(t(`${T_PATH}.required`)),
+            street_address: yup.string().required(t(`${T_PATH}.required`)),
+          }),
+        })
+        .nullable(),
+    })
+    .required();
+
+  const {
+    control,
+    handleSubmit,
+    register,
+    reset,
+    setValue,
+    watch,
+    formState: { errors, isSubmitting },
+  } = useForm<AddEditCustomerFormFields>({
+    resolver: yupResolver(schema),
+  });
+
+  const hasSecondaryProfile = watch('has_secondary_profile');
+
+  const formatDateForApi = (val: string) => {
+    return moment(val, 'D.M.YYYY', true).isValid() ? moment(val, 'D.M.YYYY').format('YYYY-MM-DD') : val;
+  };
+
+  const formatDateForInput = (val: string) => {
+    return moment(val, 'YYYY-MM-DD', true).isValid() ? moment(val, 'YYYY-MM-DD').format('D.M.YYYY') : val;
```

Comment:
Probably we don't need to check for validity here and let it throw if there's an error from the backend.

Sentry will be attached later on so it would catch that then.

Reply:
This is now updated and there's no validation